File: internal/dto/request.go
package dto

type Request struct {
	CurrPage int       `json:"currPage"`
	PageSize int       `json:"pageSize"`
	OrderBy  []orderBy `json:"orderBy"`
	Tree     bool      `json:"tree"`
	QueryBy  []queryBy `json:"queryBy"`
	Omit     string    `json:"omit"`
}

type orderBy struct {
	Col string `json:"col"`
	Asc bool   `json:"asc"`
}

type queryBy struct {
	Col string      `json:"col"`
	Val interface{} `json:"val"`
}

File: web/packages/admin/src/core/dto/request.ts
type OrderBy = {
    col: string,
    asc: boolean
}

type QueryBy = {
    col: string,
    val: any
}

export class Request {
    currPage?: number;
    pageSize?: number;
    orderBy?: OrderBy[];
    queryBy?: QueryBy[];
    omit?: string;
    tree?: boolean;

    constructor(pagination?: { currPage: number, pageSize: number }) {
        if (!!pagination) {
            this.currPage = pagination.currPage;
            this.pageSize = pagination.pageSize;
        } else {
            this.currPage = 1;
            this.pageSize = 10;
        }
    }
}
